agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileWriter.java

```diff
@@ -34,24 +34,25 @@
 /** This class manages writing a list of {@link ByteBuffer} to the file system. */
 final class LocalFileWriter {
 
-  // 50MB per folder for all apps.
-  private static final long MAX_FILE_SIZE_IN_BYTES = 52428800; // 50MB
   private static final String PERMANENT_FILE_EXTENSION = ".trn";
 
+  private final long diskPersistenceMaxSizeBytes;
   private final LocalFileCache localFileCache;
   private final File telemetryFolder;
   private final LocalStorageStats stats;
 
   private final OperationLogger operationLogger;
 
   LocalFileWriter(
+      int diskPersistenceMaxSizeMb,
       LocalFileCache localFileCache,
       File telemetryFolder,
       LocalStorageStats stats,
       boolean suppressWarnings) { // used to suppress warnings from statsbeat
     this.telemetryFolder = telemetryFolder;
     this.localFileCache = localFileCache;
     this.stats = stats;
+    this.diskPersistenceMaxSizeBytes = diskPersistenceMaxSizeMb * 1024L * 1024L;
```

Comment:
If it would not be an issue in terms of allocation, it may be worth introducing  a `DiskPersistenceMaxSize` object type.  It would have advantages:
 - A method could be added to the `DiskPersistenceMaxSize` object type to encapsulate this kind of code: `diskPersistenceMaxSizeMb * 1024L * 1024L;`
 - This would remove the need to add comments (for example in `AzureMonitorExporterBuilder` for this PR)
- This would strengthen the code checks at compilation

Reply:
comment is only for default value. creating an object for a primitive type with super simple arithmetic is not worthy in my opinion.